### PR TITLE
Updating Grafana role for enhancements

### PR DESCRIPTION
**Pull request by strus38**
A huge enhancement for the grafana role to allow:
- customizing the grafane deployment with port, dashboards, ldap, etc...

### CODE REVIEW

File: roles/grafana/README.md [hunk 1]
```diff
@@ -3,11 +3,124 @@
...
+    type: prometheus
+    access: proxy
+    url: 'http://{{ prometheus_web_listen_address }}'
+    basicAuth: false
+```
+
+Dashboard example:
+
+```yaml
+grafana_dashboards:
```
Comment (oxedions) [inline roles/grafana/README.md:77]: I tried dashboards. In fact, the role imported my json files I put i my **dashboards** folder, but ignored the ids list here.

So importing json is working, but not really documented, and importing id does not seems to work.

**Review by oxedions — CHANGES_REQUESTED** [review]
Dear @strus38 

Many thanks for this massive and impressive dev. 😊 

I tested the role basic features (installing grafana, setting user/pass, firewall, adding a dashboard, adding a datasource) with success, but had issues with dashboards. See my comments on that point.

I also think it could be interesting in the README to point somewhere to https://grafana.com/grafana/download so users know where to download it.

Also, could you change the owner of the role and become the owner ?

Regarding automatic tests, since I am way to late to provide them, I understand you added them in this PR, and I am ok with that.

**Comment by strus38** [comment]
Fixed issues... working like a charm 👍

**Review by oxedions — APPROVED** [review]
Dear @strus38 

Thanks a lot for this PR.

It is ok for me 😊

**Comment by strus38** [comment]
> Dear @strus38
> 
> Thanks a lot for this PR.
> 
> It is ok for me 😊

I will try to fix the automated tests... not sure why some are failing.

**Comment by strus38** [comment]
Finally got the CICD successful ... you can re-approve!! @oxedions  ... sorry for all that, but the only way to validate is to run it by committing more fixes :-)

**Comment by oxedions** [comment]
I understand 😊 

Restarting checks.

**Comment by oxedions** [comment]
All green, merged. Many thanks 😉

**Pull request merged**